In [1]:
import pandas as pd
import numpy as np

# 1. Importing Data

In [2]:
raw1 = pd.read_csv('bestsellers with categories.csv')

In [3]:
raw1.describe()

User Rating       Reviews       Price         Year
count   550.000000    550.000000  550.000000   550.000000
mean      4.618364  11953.281818   13.100000  2014.000000
std       0.226980  11731.132017   10.842262     3.165156
min       3.300000     37.000000    0.000000  2009.000000
25%       4.500000   4058.000000    7.000000  2011.000000
50%       4.700000   8580.000000   11.000000  2014.000000
75%       4.800000  17253.250000   16.000000  2017.000000
max       4.900000  87841.000000  105.000000  2019.000000

### So we have a total of 550 reviews in our dataset

In [4]:
raw1.shape

(550, 7)

#### 550 rows and 7 columns to be precise.

In [5]:
raw1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 550 entries, 0 to 549
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Name         550 non-null    object 
 1   Author       550 non-null    object 
 2   User Rating  550 non-null    float64
 3   Reviews      550 non-null    int64  
 4   Price        550 non-null    int64  
 5   Year         550 non-null    int64  
 6   Genre        550 non-null    object 
dtypes: float64(1), int64(3), object(3)
memory usage: 30.2+ KB


In [6]:
raw1.nunique()

Name           351
Author         248
User Rating     14
Reviews        346
Price           40
Year            11
Genre            2
dtype: int64

#### Details of Unique Author/Books name

In [7]:
raw1.head(5)

Name  \
0                      10-Day Green Smoothie Cleanse   
1                                  11/22/63: A Novel   
2            12 Rules for Life: An Antidote to Chaos   
3                             1984 (Signet Classics)   
4  5,000 Awesome Facts (About Everything!) (Natio...   

                     Author  User Rating  Reviews  Price  Year        Genre  
0                  JJ Smith          4.7    17350      8  2016  Non Fiction  
1              Stephen King          4.6     2052     22  2011      Fiction  
2        Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3             George Orwell          4.7    21424      6  2017      Fiction  
4  National Geographic Kids          4.8     7665     12  2019  Non Fiction

In [8]:
raw1.drop(['Author'],axis=1, inplace=True)

In [9]:
raw1.head(5)

Name  User Rating  Reviews  \
0                      10-Day Green Smoothie Cleanse          4.7    17350   
1                                  11/22/63: A Novel          4.6     2052   
2            12 Rules for Life: An Antidote to Chaos          4.7    18979   
3                             1984 (Signet Classics)          4.7    21424   
4  5,000 Awesome Facts (About Everything!) (Natio...          4.8     7665   

   Price  Year        Genre  
0      8  2016  Non Fiction  
1     22  2011      Fiction  
2     15  2018  Non Fiction  
3      6  2017      Fiction  
4     12  2019  Non Fiction

In [10]:
raw1.drop(['Name'],axis=1,inplace=True)

In [11]:
raw1.head(5)

User Rating  Reviews  Price  Year        Genre
0          4.7    17350      8  2016  Non Fiction
1          4.6     2052     22  2011      Fiction
2          4.7    18979     15  2018  Non Fiction
3          4.7    21424      6  2017      Fiction
4          4.8     7665     12  2019  Non Fiction

In [12]:
raw1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 550 entries, 0 to 549
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   User Rating  550 non-null    float64
 1   Reviews      550 non-null    int64  
 2   Price        550 non-null    int64  
 3   Year         550 non-null    int64  
 4   Genre        550 non-null    object 
dtypes: float64(1), int64(3), object(1)
memory usage: 21.6+ KB


In [13]:
raw1['Genre']=raw1['Genre'].replace("Non Fiction", "0")
raw1['Genre']=raw1['Genre'].replace("Fiction", "1")

In [14]:
raw1.head(5)

User Rating  Reviews  Price  Year Genre
0          4.7    17350      8  2016     0
1          4.6     2052     22  2011     1
2          4.7    18979     15  2018     0
3          4.7    21424      6  2017     1
4          4.8     7665     12  2019     0

In [15]:
raw1['Genre']=pd.to_numeric(raw1['Genre'],errors='coerce')

In [16]:
raw1.rename(columns={"Review": "No of Reviews", 
                     "Price":"Price of Book in Dollars"},inplace=True)

In [17]:
raw1.head(5)

User Rating  Reviews  Price of Book in Dollars  Year  Genre
0          4.7    17350                         8  2016      0
1          4.6     2052                        22  2011      1
2          4.7    18979                        15  2018      0
3          4.7    21424                         6  2017      1
4          4.8     7665                        12  2019      0

In [18]:
raw1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 550 entries, 0 to 549
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   User Rating               550 non-null    float64
 1   Reviews                   550 non-null    int64  
 2   Price of Book in Dollars  550 non-null    int64  
 3   Year                      550 non-null    int64  
 4   Genre                     550 non-null    int64  
dtypes: float64(1), int64(4)
memory usage: 21.6 KB


In [19]:
final_df=raw1

In [20]:
final_df.to_csv('Final_data_Amazon_Books.csv')